File: product/conference/src/main/java/ee/ctob/access/RoomDAO.java
package ee.ctob.access;

import ee.ctob.data.access.BaseRoomDAO;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface RoomDAO extends BaseRoomDAO {
    @Query(
            value = "SELECT location FROM backoffice.rooms " +
                    "WHERE room_uuid = ?1",
            nativeQuery = true
    )
    String getRoomLocationByRoomId(UUID roomUUID);
}

File: product/conference/src/main/kotlin/ee/ctob/access/RoomDAO.kt
package ee.ctob.access

import ee.ctob.data.access.BaseRoomDAO
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Transactional
@Repository
interface RoomDAO : BaseRoomDAO {

    @Query(
        value = "SELECT location FROM backoffice.rooms " +
                "WHERE room_uuid = ?1",
        nativeQuery = true
    )
    fun getRoomLocationByRoomId(roomUUID: UUID): String
}